### PR TITLE
3.x: Upgrade to Intel MPI Library 2021.4.0.441

### PR DESCRIPTION
**Pull request by enrico-usai**
### Description of changes
Intel MPI is now part of the OneAPI package. The standalone offline installer is available [here](https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-standalone-components.html#mpi).

Relevant changes:
- The installation path doesn't contain the patch version `.441`
- The EULA file is available in the `/opt/intel/licensing` path
- The base modules folder has been renamed from `impi` to `mpi`
- Installer options have been changed
- Extracted folder can be automatically deleted through the `--remove-extracted-files` flag
- Added Qt source code file to comply with https://www.qt.io/licensing/open-source-lgpl-obligations

### Tests
Integ tests:
* test_efa[sa-east-1-c5n.18xlarge-alinux2-slurm]
* test_efa[us-west-2-c6gn.16xlarge-ubuntu1804-slurm]
* test_efa[us-west-2-c6gn.16xlarge-ubuntu2004-slurm] 
* test_efa[us-west-2-c6gn.16xlarge-alinux2-slurm]
* test_efa[sa-east-1-c5n.18xlarge-alinux2-slurm] 
* test_efa[us-west-2-c6gn.16xlarge-ubuntu1804-slurm]
* test_efa[us-west-2-c6gn.16xlarge-ubuntu2004-slurm] 
* test_efa[us-west-2-c6gn.16xlarge-alinux2-slurm]

Manual tests on Alinux2. Manual execution of the `intel_mpi.rb` recipe then verified modules:
```
$ module avail
...
----------------- /opt/intel/mpi/2021.4.0/modulefiles/ -----------------
intelmpi

$ module load intelmpi
Loading mpi version 2021.4.0
$ mpicc --version
gcc (GCC) 7.3.1 20180712 (Red Hat 7.3.1-13)
Copyright (C) 2017 Free Software Foundation, Inc.
```

### References
* IntelMPI licensing (OneApi section): https://www.intel.com/content/www/us/en/developer/articles/license/end-user-license-agreement.html
* Intel MPI download: https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-standalone-components.html#mpi
* Modules documentation: https://modules.readthedocs.io/en/latest/module.html

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ This file is used to list changes made in each version of the AWS ParallelCluste
 - Upgrade NVIDIA driver to version 470.82.01.
 - Upgrade CUDA library to version 11.4.3.
 - Upgrade NVIDIA Fabric manager to version 470.82.01.
+- Upgrade Intel MPI Library to 2021.4.0.441.
```
Comment (gmarciani) [inline CHANGELOG.md:23]: We are also setting the QT version for Intel MPI to 5.15.2.
Is it worth it to mention in the changelog?

File: cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb [hunk 2]
```diff
@@ -17,40 +17,46 @@
...
 
-intelmpi_modulefile = "/opt/intel/impi/#{node['cluster']['intelmpi']['version']}/intel64/modulefiles/intelmpi"
-intelmpi_installer_archive = "l_mpi_#{node['cluster']['intelmpi']['version']}.tgz"
-intelmpi_installer_path = "#{node['cluster']['sources_dir']}/#{intelmpi_installer_archive}"
-intelmpi_installer_url = "https://#{node['cluster']['region']}-aws-parallelcluster.s3.#{node['cluster']['region']}.#{aws_domain}/archives/impi/#{intelmpi_installer_archive}"
+intelmpi_installation_path = "/opt/intel/mpi/#{node['cluster']['intelmpi']['version']}"
+intelmpi_modulefile = "#{intelmpi_installation_path}/modulefiles/intelmpi"
+intelmpi_installer = "l_mpi_oneapi_p_#{node['cluster']['intelmpi']['full_version']}_offline.sh"
+intelmpi_installer_path = "#{node['cluster']['sources_dir']}/#{intelmpi_installer}"
+intelmpi_installer_url = "https://#{node['cluster']['region']}-aws-parallelcluster.s3.#{node['cluster']['region']}.#{aws_domain}/archives/impi/#{intelmpi_installer}"
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:24]: Here we are using the naming `impi`. Shouldn't it be `mpi`?

Comment (gmarciani) [reply cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:24]: Probably it's not worth it to rename it.

File: cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb [hunk 3]
```diff
@@ -17,40 +17,46 @@
...
+    rm -f #{intelmpi_installer}
   INTELMPI
-  creates "/opt/intel/impi/#{node['cluster']['intelmpi']['version']}"
+  creates intelmpi_installation_path.to_s
 end
 
 append_if_no_line "append intel modules file dir to modules conf" do
   path node['cluster']['modulepath_config_file']
-  line "/opt/intel/impi/#{node['cluster']['intelmpi']['version']}/intel64/modulefiles/"
+  line "#{intelmpi_installation_path}/modulefiles/"
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:48]: Further improvement (we could also do it later on): determine the directory from the variable `intelmpi_modulefile `

**Review by gmarciani — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (enrico-usai) [reply CHANGELOG.md:23]: I don't think so it's just a third party dependency installed by Intel MPI itself.

File: cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb (hunk 2 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:24]: To change it we need to move artifacts in a different S3 folder and we cannot move old ones so it's better to keep all IntelMPI artifacts together.

File: cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb (hunk 3 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb:48]: I'm skipping this for now to avoid re-run of build image

**Review by gmarciani — APPROVED** [review]

**Pull request merged**